tests/components/config/test_entity_registry.py

```diff
@@ -144,6 +147,162 @@ class Unserializable:
+                area_id="area52",
+                device_id="device123",
+                entity_id="test_domain.nameless",
+                has_entity_name=True,
+                original_name=None,
+                platform="test_platform",
+                unique_id="2345",
+            ),
+            "test_domain.renamed": RegistryEntry(
+                area_id="area52",
+                device_id="device123",
+                entity_id="test_domain.renamed",
+                has_entity_name=True,
+                name="User name",
+                original_name="Hello World",
+                platform="test_platform",
+                unique_id="3456",
+            ),
+            "test_domain.boring": RegistryEntry(
+                entity_id="test_domain.boring",
+                platform="test_platform",
+                unique_id="4567",
+            ),
+            "test_domain.disabled": RegistryEntry(
+                disabled_by=RegistryEntryDisabler.USER,
+                entity_id="test_domain.disabled",
+                hidden_by=RegistryEntryHider.USER,
+                platform="test_platform",
+                unique_id="789A",
+            ),
+            "test_domain.hidden": RegistryEntry(
+                entity_id="test_domain.hidden",
+                hidden_by=RegistryEntryHider.USER,
+                platform="test_platform",
+                unique_id="89AB",
+            ),
+            "sensor.default_precision": RegistryEntry(
+                entity_id="sensor.default_precision",
+                options={"sensor": {"suggested_display_precision": 0}},
+                platform="test_platform",
+                unique_id="9ABC",
+            ),
+            "sensor.user_precision": RegistryEntry(
+                entity_id="sensor.user_precision",
+                options={
+                    "sensor": {"display_precision": 0, "suggested_display_precision": 1}
+                },
+                platform="test_platform",
+                unique_id="ABCD",
+            ),
+        },
+    )
+
+    await client.send_json_auto_id({"type": "config/entity_registry/list_for_display"})
+    msg = await client.receive_json()
+
+    assert msg["result"] == {
+        "entity_categories": {"0": "config", "1": "diagnostic"},
+        "entities": [
+            {
+                "ai": "area52",
+                "di": "device123",
+                "ec": 1,
+                "ei": "test_domain.test",
+                "en": "Hello World",
+                "tk": "translations_galore",
+            },
+            {
+                "ai": "area52",
+                "di": "device123",
+                "ei": "test_domain.nameless",
+                "en": None,
+            },
+            {
+                "ai": "area52",
+                "di": "device123",
+                "ei": "test_domain.renamed",
+            },
+            {
+                "ei": "test_domain.boring",
+            },
```

Comment:
Maybe a silly question but… do we still need this entry since it has no useful information?

Reply:
I think we need it so frontend knows that the entity is in the entity registry and can be customized?

Reply:
This API output is only meant for displaying. When we open more info and the settings screen will be shown, it's based on the other output. 

CC @bramkragten @piitaya

Reply:
Yes, the config-entities page can use `list` endpoint.